app/Handlers/TitleRewritingHandler.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+class FreshRSS_TitleRewriting_Handler {
+	const TOKEN_VARIABLE_START = '{';
+	const TOKEN_VARIABLE_END = '}';
+	const TOKEN_VARIABLE_FILTER = '|';
+	const TOKEN_VARIABLE_FILTER_PARAM_START = '(';
+	const TOKEN_VARIABLE_FILTER_PARAM_END = ')';
+	const TOKEN_VARIABLE_FILTER_PARAM_DELIMITER = ',';
+	const TOKEN_SPACE = ' ';
+	const TOKEN_PARAM_STRING_DELIMITER = '"';
+
+	const SUPPORTED_VARIABLES = [
+		'title',
+		'feed',
+	];
+	const SUPPORTED_FILTERS = [
+		[
+			'name' => 'trim',
+			'parameters' => [
+				0,
+				1,
+			],
+		],
+		[
+			'name' => 'replace',
+			'parameters' => [
+				2
+			],
+		],
+		[
+			'name' => 'ireplace',
+			'parameters' => [
+				2
+			],
+		],
+	];
+
+	private $rule = [];
+
+	public function __construct(string $rule) {
+		$this->parseRule($rule);
+		$this->cleanRule();
+	}
+
+	private function parseRule(string $rule) {
```

Comment:
Please, split this method for better readability and maintainability

Reply:
I am not sure if it will make this more readable. I'll try something.